code/modules/busy_space/air_traffic.dm

```diff
@@ -0,0 +1,128 @@
+						"current solar weather info" = list("sending you the relevant information via tightbeam", "cannot fulfill your request at the moment"),
+						"nearby traffic info" = list("sending you current traffic info", "no available info in your area"),
+						"remote telemetry data" = list("sending telemetry now", "no uplink from your ship, recheck your uplink and ask again"),
+						"refueling information" = list("sending refueling information now", "no fuel for your ship class in this sector"),
+						"a current system time sync" = list("sending time sync ping to you now", "your ship isn't compatible with our time sync, set time manually"),
+						"current system starcharts" = list("transmitting current starcharts", "your request is queued, overloaded right now"),
+						"permission to engage FTL" = list("permission to engage FTL granted, good day", "permission denied, wait for current traffic to pass"),
+						"permission to transit system" = list("permission to transit granted, good day", "permission denied, wait for current traffic to pass"),
+						"permission to depart system" = list("permission to depart granted, good day", "permission denied, wait for current traffic to pass"),
+						"permission to enter system" = list("good day, permission to enter granted", "permission denied, wait for current traffic to pass"),
+						)
+
+	//Random chance things for variety
+	var/chatter_type = "normal"
+	if(force_chatter_type)
+		chatter_type = force_chatter_type
+	else
+		chatter_type = pick(2;"emerg",5;"wrong_freq","normal") //Be nice to have wrong_lang...
+
+	var/yes = prob(90) //Chance for them to say yes vs no
+
+	var/request = pick(requests)
+	var/callname = pick(alt_atc_names)
+	var/response = requests[request][yes ? 1 : 2] //1 is yes, 2 is no
+
+	var/full_request
+	var/full_response
+	var/full_closure
+
+	switch(chatter_type)
+		if("wrong_freq")
+			callname = pick(wrong_atc_names)
+			full_request = "[callname], this is [combined_name] on a [mission] [pick(mission_noun)] to [destname], [pick(request_verb)] [request]."
+			full_response = "[combined_name], this is [using_map.station_short] TraCon, wrong frequency. Switch to [rand(700,999)].[rand(1,9)]."
+			full_closure = "[using_map.station_short] TraCon, understood, apologies."
+		if("wrong_lang")
+			//Can't implement this until autosay has language support
+		if("emerg")
+			var/problem = pick("hull breaches on multiple decks","unknown life forms on board","a drive about to go critical","asteroids impacting the hull","a total loss of engine power","people trying to board the ship")
+			full_request = "This is [combined_name] declaring an emergency! We have [problem]!"
+			full_response = "[combined_name], this is [using_map.station_short] TraCon, copy. Switch to emergency responder channel [rand(700,999)].[rand(1,9)]."
+			full_closure = "[using_map.station_short] TraCon, okay, switching now."
```

Comment:
roger correctly used means "i understood and will do as instructed", although it is used a lot incorrectly for example as you say.

Also, uber correctly, a frequency hand off needs to be acknowledged, so the readback would be "contacting _frequency above_"